libdnf/dnf-keyring.cpp

```diff
@@ -252,9 +278,27 @@ dnf_keyring_check_untrusted_file(rpmKeyring keyring,
         goto out;
     }
 
-    /* we don't want to abort on missing keys */
     ts = rpmtsCreate();
-    rpmtsSetVSFlags(ts, _RPMVSF_NOSIGNATURES);
+
+    if (rpmtsSetKeyring(ts, keyring) < 0) {
+        g_set_error_literal(error, DNF_ERROR, DNF_ERROR_INTERNAL_ERROR, "failed to set keyring");
+        goto out;
+    }
+    rpmtsSetVfyLevel(ts, RPMSIG_SIGNATURE_TYPE | RPMSIG_DIGEST_TYPE);
```

Comment:
You might want to leave out RPMSIG_DIGEST_TYPE here - digests aka hashes are not signatures, and this is about checking signatures. A practical consequence of enforcing digests here would be packages built with rpm < 4.14 failing the check in FIPS mode (RHEL 8 already) but probably elsewhere too in not too distant future due to MD5. Such packages can still be verified using the signature (assuming non-obsolete crypto is used for signing), but this would require disabling signature checking entirely (on libdnf level) to get past it.

Reply:
I see, thank you,
I have removed it.